### ISSUE TITLE
Training student model on pseudo-labeled real images

### ISSUE THREAD

**Issue by luochunfen** [DepthAnything/Depth-Anything-V2#108]
Thanks your great job ! Could you give more details about discarding 10% largest-loss regions when training student models on pseudo-labeled real images ? I can't figure out how to implement this trick.

**Comment by LiheYoung** [comment]
Hi, suppose you have a predicted depth map and a groundtruth depth map, both of shape HxW. You can calculate their pixel-wise loss to get the loss map `L` of HxW. Then you can use some Library functions such as PyTorch `percentile` to obtain the 10%-th largest value `v` in this loss map. Then you can filter the original loss map by `loss = L[L < v].mean()`.

**Comment by luochunfen** [comment]
> Hi, suppose you have a predicted depth map and a groundtruth depth map, both of shape HxW. You can calculate their pixel-wise loss to get the loss map `L` of HxW. Then you can use some Library functions such as PyTorch `percentile` to obtain the 10%-th largest value `v` in this loss map. Then you can filter the original loss map by `loss = L[L < v].mean()`.

I got it. Thank you !!!

**Comment by smilebetterworld** [comment]
@LiheYoung , so predicted depth map is student model predicted depth, and  groundtruth depth map is pseudo-labeled from teacher model infer?
